app/lib/features/space/actions/update_feature_power_level.dart

```diff
@@ -0,0 +1,249 @@
+import 'package:acter/common/toolkit/buttons/primary_action_button.dart';
+import 'package:acter/features/space/settings/pages/apps_settings_page.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_easyloading/flutter_easyloading.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:logging/logging.dart';
+
+final _log = Logger('a3::space::settings::update_feature_level');
+
+Future<bool> updateFeatureLevelChangeDialog(
+  BuildContext context,
+  int maxPowerLevel,
+  int? currentPw,
+  Space space,
+  RoomPowerLevels powerLevels,
+  String levelKey,
+  String featureName,
+) async {
+  final lang = L10n.of(context);
+  final newPowerLevel = await showDialog<int?>(
+    context: context,
+    builder: (BuildContext context) => _ChangePowerLevelDialog(
+      featureName: featureName,
+      currentPowerLevelName:
+          maxPowerLevel == 100 ? powerLevelName(currentPw) : 'Custom',
+      currentPowerLevel: currentPw,
+    ),
+  );
+  if (newPowerLevel == currentPw) return false;
+  if (!context.mounted) {
+    EasyLoading.dismiss();
+    return true;
+  }
+
+  return await updateFeatureLevel(
+    lang,
+    space,
+    levelKey,
+    featureName,
+    newPowerLevel,
+  );
+}
+
+Future<bool> updateFeatureLevel(
+  L10n lang,
+  Space space,
+  String levelKey,
+  String featureName,
+  int? newPowerLevel,
+) async {
+  EasyLoading.show(status: lang.changingSettingOf(featureName));
+  try {
+    final res = await space.updateFeaturePowerLevels(
+      levelKey,
+      newPowerLevel,
+    );
+    EasyLoading.showToast(lang.powerLevelSubmitted(featureName));
+    return res;
+  } catch (e, s) {
+    _log.severe('Failed to change power level of $featureName', e, s);
+    EasyLoading.showError(
+      lang.failedToChangePowerLevel(e),
+      duration: const Duration(seconds: 3),
+    );
+    return false;
+  }
+}
+
+class _ChangePowerLevelDialog extends StatefulWidget {
+  final String featureName;
+  final int? currentPowerLevel;
+  final String currentPowerLevelName;
+
+  const _ChangePowerLevelDialog({
+    required this.featureName,
+    required this.currentPowerLevelName,
+    this.currentPowerLevel,
+  });
+
+  @override
+  State<_ChangePowerLevelDialog> createState() =>
+      __ChangePowerLevelDialogState();
+}
+
+class __ChangePowerLevelDialogState extends State<_ChangePowerLevelDialog> {
+  final TextEditingController dropDownMenuCtrl = TextEditingController();
+  final GlobalKey<FormState> _formKey =
+      GlobalKey<FormState>(debugLabel: 'change power level form');
+
+  String? currentMemberStatus;
+  int? customValue;
+
+  @override
+  void initState() {
+    super.initState();
+    setState(() => currentMemberStatus = widget.currentPowerLevelName);
```

Comment:
No this should be not here. During widget initialization, ```setState``` cannot be called as it's already building. This should throw exception on widget ```build()```...
It should be wrapped with ```PostFrameCallback```.

Reply:
this is a copy-paste from the power-level dialog, where it works just fine. I can fix it, but this isn't throwing any errors.

Reply:
Having it not scheduled for next frame would have no effect as you can see the comment [here](https://stackoverflow.com/a/53373017). So it'd just work fine without wrapping it with ```setState```.